docs/elasticsearch-spec.asciidoc

```diff
@@ -359,64 +359,51 @@ The init container inherits the image of the main container image if one is not
-
-[id="{p}-change-budget"]
-==== Change budget
+[source,yaml]
+----
+spec:
+  updateStrategy:
+    changeBudget:
+      maxSurge: 3
+      maxUnavailable: 1
+----
 
-No downtime should be expected when the cluster topology changes. Shards on deprecated nodes are migrated away so the node can be safely removed.
+`maxSurge` refers to the number of Pods that can be scheduled above the total number of Pods in the currently applied spec.
+`maxUnavailable` refers to the number of Pods that can be unavailable out of the total number of Pods in the currently applied spec. Unavailable is defined as Pod being not ready from Kubernetes perspective.
 
-For example, to mutate a 3-nodes cluster with 16GB memory limit on each node to a 3-nodes cluster with 32GB memory limit on each node, ECK will:
+Note that these constraints are enforced immediately after the spec is applied, which might mean that the actual state of the cluster is in violation of these constraints. This is expected and the operator won't stop from moving the cluster towards respecting them.
+For example, if a new spec defines a larger cluster with no unavailable Pods permitted (`maxUnavailable: 0`), the operator will continue creating missing Pods according to the best practices. Similarly, if a new spec defines a smaller cluster with no Pods above the spec permitted (`maxSurge: 0`), the operator will continue safely removing unnecessary Pods.
 
-1. Add a new 32GB node: the cluster temporarily has 4 nodes
-2. Migrate data away from the first 16GB node
-3. Once data is migrated, remove the first 16GB node
-4. Follow the same steps for the 2 other 16GB nodes
+==== Specifying changeBudget
+For both `maxSurge` and `maxUnavailable` you can specify the following values:
 
-The cluster health stays green during the entire process.
-By default, only one extra node can be added on top of the expected ones. In the example above, a 3-nodes cluster may temporarily be composed of 4 nodes while data migration is in progress.
+* `null` - the default value will be used,
+* non-negative - the value will be used as is,
+* negative - the value will be unbounded.
 
-This behaviour can be controlled through the `changeBudget` section of the cluster specification `updateStrategy`. If not specified, it defaults to the following:
+==== Default behavior
+When `updateStrategy` is not present in the spec, it defaults to the following:
 
 [source,yaml]
 ----
 spec:
   updateStrategy:
     changeBudget:
-      maxSurge: 1
-      maxUnavailable: 0
+      maxSurge: -1
+      maxUnavailable: 1
 ----
+`maxSurge` is unbounded. This results in immediate creation of all missing Pods that best practices allow for at the given time.
+`maxUnavailable` defaults to `1`. This assures that cluster will have at most one unavailable Pod at any given time.
```

Comment:
`maxSurge` is unbounded: This means that all the missing Pods are immediately created according to the best practices at any given time.
  `maxUnavailable` defaults to `1`: This assures that clusters have at most one unavailable Pod at any given time.